pkg/kindsys/report.go

```diff
@@ -54,17 +73,53 @@ var plannedCoreKinds = []string{
 	"QueryHistory",
 }
 
+type Kind struct {
+	kindsys.SomeKindProperties
+	Category  string
+	SchemaRef string
+	GoRef     string
+	TsRef     string
+	DocsRef   string
+}
+
+// MarshalJSON is overwritten to marshal
+// kindsys.SomeKindProperties at root level.
+func (k Kind) MarshalJSON() ([]byte, error) {
```

Comment:
Any kind of suggestion to make this bit more beautify will be more than welcome. 
I just spent the enough time to write a working version!